### ISSUE TITLE
Introduce an ability to build `AggregatedProjection`s from regular `Projection`s.

### ISSUE THREAD

**Issue by armiol** [SpineEventEngine/core-java#664]
As of now there is an ability to define a number of `Projection`s, which are asynchronously updated according to the events, produced by the write-side. It works fine for a majority of cases.

However, it's rather difficult to implement several inter-related `Projection` types. To illustrate the case, let's imagine there is a number of reports, calculating the number of user actions in some domain. And according to the requirements, there are three reports, which should be up-to-date at any time:

* `SumByHour` — total number of actions performed during a particular hour (e.g. 3 PM - 4 PM of January 7, 2018 day).
* `SumByDay` — number of daily actions (e.g. for 12 AM - 11:59 PM of January 7, 2018).
* `SumByMonth` — number of actions per month (e.g. 1 to 31 of January 2018).

If those are three regular projections, they consume the same events — as they count those. However, projections may be served by the different application nodes. In this case due to transport and protocol considerations, the "live" events are delivered to projection instances out of sync making today numbers for the hour/day/month inconsistent.

The idea of this feature request is to introduce another type of `Entity`, called `AggregatedProjection`. Its features are as follows:

* Have a particular state record similar to what regular `Projection` has — as opposed to `Aggregate`, which always loads from its events.
* Subscribe to other `Projection` (**TBD: maybe, all kinds of entities?**) state updates and build own state from them.
* Expose own state to `QueryService` similar to `Projection`s.

In the imagined example above, `SumByDay` (an `AggregatedProjection`) could be built from `SumByHour` (regular `Projection`).

In code that could look like this:

```java

public class SumByDayView extends AggregatedProjection<DayId, SumByDay> {

    @Subscribe
    void on(SumByHour sumByHour) {
        // calls to {@code getBuilder()} and updates its state.
    }
}
```

In order this feature to be implemented we should consider the following points:

* Added message routing complexity and thus more difficult scaling.
* The fact that regular `Projection`s may have this feature as well — and subscribe to `Entity` state updates as well as to `Event`s. However, this would make `Projections` much more complex and potentially would encourage the framework users to turn them into "God" objects. 
* This feature also rhymes with the idea to turn Stand into a `SubscriptionBus` (see #608) and should probably be implemented after it.

**Comment by armiol** [comment]
@alexander-yevsyukov FYI.

**Comment by armiol** [comment]
Some of the features are going to be implemented in #733.

**Comment by armiol** [comment]
It is now possible to implement by subscribing to the entity state updates.
